docs/work-flow.md

```diff
@@ -0,0 +1,176 @@
+# DC/OS Development Workflow
+
+
+        "... there is absolutely nothing "obvious" about source-control systems, or workflows before you already know them"
+
+Mesosphere DC/OS has two primary repositories.
+
+1. Open Source DC/OS (https://github.com/dcos/dcos)
+2. Closed Source Mesosphere Enterprise DC/OS (https://github.com/mesosphere/dcos-enterprise)
+
+In our design, the closed source Mesosphere Enterprise DC/OS depends on the Open Source DC/OS, [builds on top of the artifact](https://github.com/mesosphere/dcos-enterprise/blob/master/util/build.py) from the Open source DC/OS.
+
+We have designed our DC/OS pull request workflow, in such a way,  we test our Open Source changes
+with the Enterprise consumer,  we keep both versions ready for release at any moment in time.
+It can be cumbersome, and error-prone to deal with two version control systems at the same time.
+A friendly bot called `@mesosphere-mergebot` comes to our rescue and helps the developers with the process.
+
+#### How do I contribute a change to dcos/dcos ?
+
+If a change is only to Enterprise DC/OS, the developer should create the pull request against the Enterprise Repository.
+
+For all contributions, a pull request is against Open Source DC/OS and tested with both Open and
+Enterprise, and we merge the Open Source pull request, and update the reference in Enterprise.
+
+#### What should be my target branch?
+
+All Pull Requests should be against `master` branch, and the bug-fixes should be actively backported to the
+supported versions of DC/OS. The dcos/dcos repo will have the list of actively supported branches, and developers
+should create the back-port pull requests if they have to include the change in the particular version DC/OS.
+
+#### What should I do after I create a pull request against DC/OS?
+
+We use labels, to set the status of a pull-request. All pull requests start at "Work In
+Progress" state. The pull request creator can continue working on this until he/she is ready.
+
+When a pull request is ready for review, the pull request creator is expected to run this command.
+
+```
+@mesosphere-mergebot label ready for review
+```
+
+The above command will set the label of the PR to "Ready For Review" and it is crucial,
+as it will now fall under the radar of Mergebot to monitor for status updates to the PR.
+
+And also, do update to Enterprise reference using the command:
```

Comment:
"do update to Enterprise reference" - I'm not sure exactly what you mean by this. But I imagine something like.

"Each Pull Request to Open Source DC/OS is matched by a Pull Request to DC/OS Enterprise. Use the following command to create or update that matching Pull Request:"